### PR TITLE
Refactor trafficlight adapter

### PR DESCRIPTION
**Pull request by MidhunSureshR**
- [x] Move actions to separate files (eg: register, login, logout in auth.ts)
- [ ] Report errors to trafficlight
- [x] Make sure IDE picks up custom command

### CODE REVIEW

File: cypress/e2e/trafficlight/trafficlight.spec.ts [hunk 1]
```diff
@@ -70,232 +132,70 @@ function recurse() {
...
-            cy.get(".mx_AccessibleButton.mx_BaseCard_back").click();
-            cy.get(".mx_AccessibleButton.mx_BaseCard_close", { timeout: 30000 }).click();
-            return "invited";
-        }
-        case "open-room": {
-            cy.get(".mx_RoomSublist_tiles").contains(data["name"]).click();
-            return "room-opened";
-        }
+            return changeRoomHistoryVisibility(data);
+        case "open-room":
```
Comment (michaelkaye) [inline cypress/e2e/trafficlight/trafficlight.spec.ts:164]:
```suggestion
        case "open-room":
        case "open_room":
```

Should we standardise on `this_case` as the format for action names - we can handle both at once as a temporary measure?

File: cypress/e2e/trafficlight/trafficlight.spec.ts [hunk 2]
```diff
@@ -70,232 +132,70 @@ function recurse() {
...
-                cy.get('.mx_CreateRoomDialog_topic input').type(data['topic']);
-            }
-            // do this to prevent https://github.com/vector-im/element-web/issues/22590, weirdly
-            // cy.get('.mx_CreateRoomDialog_name input').click();
-            // cy.wait(5000);
+            return verifyDeviceIsTrusted();
+        case "enable_dehydrated_device":
+            return enableDehydratedDevice(data);
+        case "enable_key_backup":
+            return enableKeyBackup(data);
```
Comment (michaelkaye) [inline cypress/e2e/trafficlight/trafficlight.spec.ts:155]:
```suggestion
            const password = data["key_backup_passphrase"];
            if (!password) {
                    throw new Error("'key_backup_passphrase' not in data for action 'enable_dehydrated_device'");
            }
            return enableKeyBackup(password);
```

Perhaps this style so we can marshall/unmarshall the json here, and have just the control of the client in the method?

File: cypress/e2e/trafficlight/actions/e2ee.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    cy.gotoAllSettings();
+    cy.get("[data-testid='settings-tab-USER_SECURITY_TAB']").click();
+    cy.get(".mx_SecureBackupPanel_buttonRow").contains("Set up").click();
+    cy.get(".mx_CreateSecretStorageDialog_optionIcon_securePhrase").click();
+    cy.get(".mx_CreateSecretStorageDialog [data-testid='dialog-primary-button']").click();
+    const password = data["key_backup_passphrase"];
+    if (!password) {
+        throw new Error("'key_backup_passphrase' not in data for action 'enable_dehydrated_device'");
+    }
+    cy.get(".mx_CreateSecretStorageDialog_passPhraseContainer input[type='password']").type(password);
```
Comment (michaelkaye) [inline cypress/e2e/trafficlight/actions/e2ee.ts:75]:
```suggestion
export function enableKeyBackup(password: string): string {
    cy.gotoAllSettings();
    cy.get("[data-testid='settings-tab-USER_SECURITY_TAB']").click();
    cy.get(".mx_SecureBackupPanel_buttonRow").contains("Set up").click();
    cy.get(".mx_CreateSecretStorageDialog_optionIcon_securePhrase").click();
    cy.get(".mx_CreateSecretStorageDialog [data-testid='dialog-primary-button']").click();
    cy.get(".mx_CreateSecretStorageDialog_passPhraseContainer input[type='password']").type(password);
```

**Review by michaelkaye — COMMENTED** [review]
Was just looking for the open-room action. but some suggestions as i was reading the adapter

File: cypress/e2e/trafficlight/trafficlight.spec.ts (hunk 1 above)
Comment (MidhunSureshR) [reply cypress/e2e/trafficlight/trafficlight.spec.ts:164]: yes, we should choose one and mention it in the trafficlight repo as a code style. My personal preference would be `open-room` but I'm okay with either. The proxy currently uses `camcelCase`, so I'll need to change that as well.

**Comment by michaelkaye** [comment]
After some IRL discussion; We're going to merge this with the error handling uncompleted; there's future changes to this which also need to be merged so going to handle them in a separate PR.

**Pull request merged**